packages/x-components/src/x-modules/facets-next/components/lists/__tests__/filters-injection.mixin.spec.ts

```diff
@@ -0,0 +1,189 @@
+import { Filter } from '@empathyco/x-types-next';
+import { mount, Wrapper } from '@vue/test-utils';
+import Vue, { ComponentOptions } from 'vue';
+import { Component, Prop } from 'vue-property-decorator';
+import {
+  createNextHierarchicalFilter,
+  createNextSimpleFilter
+} from '../../../../../__stubs__/filters-stubs.factory';
+import { XProvide } from '../../../../../components/decorators/injection.decorators';
+import FiltersInjectionMixin from '../filters-injection.mixin';
+
+@Component({
+  template: `
+    <div>
+    <slot/>
+    </div>`
+})
+class Provider extends Vue {
+  @Prop()
+  @XProvide('filters')
+  public injectedData!: any;
+}
+
+const Filters: ComponentOptions<Vue> = {
+  mixins: [FiltersInjectionMixin],
+  template: `
+    <div>
+    <li v-for="filter of renderedFilters" :key="filter.id">{{ filter.id }}</li>
+    </div>
+  `
+};
+
+describe('Test filters injection mixin', () => {
+  function renderFiltersMixin({
+    propFilters,
+    injectedFilters,
+    parentId
+  }: Partial<{
+    propFilters: Filter[];
+    injectedFilters: Filter[];
+    parentId: Filter['id'];
+  }> = {}): TestFilterInjectionAPI {
+    const wrapper = mount(
+      {
+        template: `
+          <Provider :injectedData="injectedFilters">
+          <Filters :filters="propFilters" :parentId="parentId"/>
+          </Provider>
+        `,
+        components: { Provider, Filters }
+      },
+      {
+        data() {
+          return {
+            propFilters,
+            injectedFilters,
+            parentId
+          };
+        }
+      }
+    );
+
+    return {
+      wrapper,
+      getRenderedFilterIds(): string[] {
+        return wrapper.findAll('li').wrappers.map(element => element.text());
+      },
+      async setPropFilters(propFilters: Filter[]) {
+        await wrapper.setData({ propFilters });
+      }
+    };
+  }
+
+  it('renders the props filters', async () => {
+    const whiteFilter = createNextSimpleFilter('color', 'white');
```

Comment:
This block is repeated a couple of times, would extract it to a function be useful?

```js
[...]

const [whiteFilter, blackFilter, blueFilter] = createNextSimpleFilters();

[...]

function createGenericSimpleFilters(): SimpleFilter[] {
  return [
    createNextSimpleFilter('color', 'white'),
    createNextSimpleFilter('color', 'black'),
    createNextSimpleFilter('color', 'blue')
  ];
}
```

Reply:
I prefer to leave it explicit in each test, it is clearer to read and easier to refactor each test separately.

Reply:
I prefer to leave it explicit in each test, it is clearer to read and easier to refactor each test separatelly.